src/shims/foreign_items.rs

```diff
@@ -362,45 +393,54 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
 
             // Rust allocation
-            // (Usually these would be forwarded to to `#[global_allocator]`; we instead implement a generic
-            // allocation that also checks that all conditions are met, such as not permitting zero-sized allocations.)
             "__rust_alloc" => {
                 let &[ref size, ref align] = this.check_shim(abi, Abi::Rust, link_name, args)?;
                 let size = this.read_scalar(size)?.to_machine_usize(this)?;
                 let align = this.read_scalar(align)?.to_machine_usize(this)?;
                 Self::check_alloc_request(size, align)?;
```

Comment:
Do we really still want to do these things if we end up dispatching to a user-defined allocator? At that point I feel like we are doing a regular function call and should impose no further restrictions on the language level. So far we have refrained from doing anything library-level in Miri -- but allocators are arguably blurring the line anyway.

Reply:
Yeah, I agree, the allocation conditions shouldn't be checked for user allocators